packages/e2e/test/local-network/local-network.test.ts

```diff
@@ -108,4 +109,190 @@ describe('Local Network', () => {
     wallet1.shutdown();
     wallet2.shutdown();
   });
+
+  it('will mint an NFT using native scripts.', async () => {
+    const amountFromFaucet = 100_000_000;
+
+    const wallet = (await getWallet({ env, idx: 10, name: 'Minting Wallet', polling: { interval: 1000 } })).wallet;
+
+    const minterKeyAgent = await KeyManagement.InMemoryKeyAgent.fromBip39MnemonicWords(
+      {
+        getPassword: async () => Buffer.from(''),
+        mnemonicWords: env.KEY_MANAGEMENT_PARAMS.mnemonic.split(' '),
+        networkId: Cardano.NetworkId.testnet
+      },
+      { inputResolver: { resolveInputAddress: async () => null } }
+    );
+
+    const policyKeyDerivationPath = {
+      index: 0,
+      role: KeyManagement.KeyRole.External
+    };
+
+    const derivedAddress = await minterKeyAgent.deriveAddress({
+      index: 0,
+      type: KeyManagement.AddressType.External
+    });
+
+    await firstValueFrom(wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+
+    const [{ address: sendingAddress }] = await firstValueFrom(wallet.addresses$);
+
+    await faucetProvider.request(sendingAddress.toString(), amountFromFaucet, 1);
+
+    const keyHash = Cardano.policyKeyHash(derivedAddress.address);
+    const extraSigner = new KeyManagement.util.KeyAgentTransactionSigner(minterKeyAgent, policyKeyDerivationPath);
+
+    const script: Cardano.NativeScript = {
+      __type: Cardano.ScriptType.Native,
+      kind: Cardano.NativeScriptKind.RequireAllOf,
+      scripts: [
+        {
+          __type: Cardano.ScriptType.Native,
+          keyHash: Cardano.Ed25519KeyHash(keyHash),
+          kind: Cardano.NativeScriptKind.RequireSignature
+        }
+      ]
+    };
+
+    const policyId = Cardano.nativeScriptPolicyId(script);
+    const assetName = Cardano.AssetName('BABABABABABABA');
+    const assetId = Cardano.AssetId(policyId.toString() + assetName.toString());
+
+    // Wait until wallet is aware of the funds.
+    await firstValueFrom(wallet.balance.utxo.total$.pipe(filter(({ coins }) => coins >= amountFromFaucet)));
+
+    const txProps = {
+      extraSigners: [extraSigner],
+      mint: new Map([[assetId, 1n]]),
+      outputs: new Set([
+        {
+          address: sendingAddress,
+          value: {
+            assets: new Map([[assetId, 1n]]),
+            coins: 50_000_000n
+          }
+        }
+      ]),
+      scripts: [script]
+    };
+    const unsignedTx = await wallet.initializeTx(txProps);
+
+    const finalizeProps = {
+      extraSigners: [extraSigner],
+      scripts: [script],
+      tx: unsignedTx
+    };
+
+    const signedTx = await wallet.finalizeTx(finalizeProps);
+    await wallet.submitTx(signedTx);
+
+    // Wait until wallet is aware of the minted token.
+    await firstValueFrom(
+      wallet.balance.utxo.total$.pipe(filter(({ assets }) => (assets ? assets.has(assetId) : false)))
+    );
+
+    // Wait until wallet is aware of the burned token.
```

Comment:
This `await` seems redundant though
```suggestion
    // Wait until wallet is aware of the minted token.
```

Reply:
Agree, and also this test was just removed since our nft e2e test now covers this case